### PR TITLE
THORN-2041: Adding support for Keycloak based security for Teiid ODat…

### PR DESCRIPTION
**Pull request by rareddy**
…a interface

- [ x] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [x ] Have you created a [JIRA](https://issues.jboss.org/browse/SWARM) and used it in the commit message?
- [ x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/wildfly-swarm/wildfly-swarm/pulls) for the same issue?
- [ x] Have you built the project locally prior to submission with `mvn clean install`?

-----

### CODE REVIEW

File: fractions/teiid/odata/src/main/java/org/wildfly/swarm/odata/ODataFraction.java [hunk 1]
```diff
@@ -15,17 +15,23 @@
...
 import static org.wildfly.swarm.spi.api.Defaultable.string;
 
 import org.wildfly.swarm.spi.api.Defaultable;
 import org.wildfly.swarm.spi.api.Fraction;
 import org.wildfly.swarm.spi.api.annotations.Configurable;
 
 public class ODataFraction implements Fraction<ODataFraction> {
-
-    @Configurable("swarm.odata.context")
+    @Configurable("swarm.teiid.odata.context")
```
Comment (Ladicek) [inline fractions/teiid/odata/src/main/java/org/wildfly/swarm/odata/ODataFraction.java:26]: Might want to keep the original one for compatibility? (But add it as a _2nd_ annotation, as the 1st one is what is documented.)

Comment (rareddy) [reply fractions/teiid/odata/src/main/java/org/wildfly/swarm/odata/ODataFraction.java:26]: So, I would able to do two @Configurable annotations? Also I willing to break the compatibility for sake of correcting it.

Comment (Ladicek) [reply fractions/teiid/odata/src/main/java/org/wildfly/swarm/odata/ODataFraction.java:26]: I don't know how many users of this fraction are out there, but if you're willing to break compat, then fine :-)

Comment (rareddy) [reply fractions/teiid/odata/src/main/java/org/wildfly/swarm/odata/ODataFraction.java:26]: I don't think many yet, we are primary consumers of this in another OpenShift based UI project.

**Comment by sberyozkin** [comment]
Should the KC fraction dependency be optional/provided ?

**Comment by rareddy** [comment]
@sberyozkin It is optional as the user can choose to secure the service or not, but I was not sure how I can use the reference to the "Secured.class" in the code but make the pom.xml define the dependency as optional

**Comment by sberyozkin** [comment]
I guess having a KC integration out of the box for this fraction is fine

**Comment by rareddy** [comment]
Sure. if there is a way we can move Secured.class as a common annotation that would remove this need. If no further questions please consider a merge.

**Comment by sberyozkin** [comment]
I've been looking at adding something similar to the management console, based on your PR. There I added 'keycloak.secured' as opposed to 'secured' only given that ManagementConsole can be secured using the other mechanisms. I've added 'keycloak.role' too. I guess Teiid OData can be secured differently as well ? Do you think it can make sense to update a little bit your PR ?
FYI, I've no rights to approve

**Comment by sberyozkin** [comment]
I ended up going the different route as the Thorntail YAML config mostly works, so my comment that I added a "keycloak.secured" property is no longer true, for now at least, IMHO it might make sense to qualify it in your case too but it is up to you whether to update it, may be securing Teiid OData with KC is the only reasonable option and as such no need to do the extra qualification...thanks

**Comment by rareddy** [comment]
@sberyozkin On standalone Teiid, we supported HTTPBasic and OAuth with Keycloak, but on OpenShift we want to tighten the offering to Keycloak.

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**